openapi_core/unmarshalling/schemas/unmarshallers.py

```diff
@@ -182,59 +190,51 @@ def unmarshal(self, value):
         else:
             return self._unmarshal_object(value)
 
+    def _clone(self, schema):
+        return ObjectUnmarshaller(
```

Comment:
@sisp can you explain me more why you create new unmarshaller with new schema but with old validator and formatter?

Reply:
Good question. I believe it's better to use the [`SchemaUnmarshallersFactory`](https://github.com/p1c2u/openapi-core/blob/df4bbfe0d7b2a88e86f5f9a47d31549197717e47/openapi_core/unmarshalling/schemas/factories.py#L39-L121) instead. Not sure why I hadn't thought of that at the time.

I've created a PR: https://github.com/p1c2u/openapi-core/pull/425

Reply:
Thanks